lib/cn/cn_tree.c

```diff
@@ -916,16 +916,19 @@ cn_comp_release(struct cn_compaction_work *w)
             le = list_prev_entry(le, le_link);
         }
 
-        if (w->cw_have_token)
-            cn_node_comp_token_put(w->cw_node);
-
-        if (w->cw_action == CN_ACTION_SPLIT)
-            cn_node_comp_token_put(w->cw_tree->ct_root);
-
         /* If cn_comp_update_XXX() was called and was successful then it will
-         * have updated busycnt.  For all errors we must do it here.
+         * have updated tn_rspill_sync, busycnt, and the token.  For all errors
+         * we must do it here.
          */
+        if (w->cw_action == CN_ACTION_SPLIT) {
+            w->cw_tree->ct_root->tn_rspill_sync--;
+            w->cw_node->tn_rspill_sync = 0;
+        }
```

Comment:
We'll want to keep an eye on this with slow media.  If with slow media rspill can't keep up then probably it indicates a problem with the throttle...